code/modules/mob/living/carbon/slime/slime.dm

```diff
@@ -204,6 +204,12 @@
 	updatehealth()
 	return
 
+/mob/living/carbon/slime/MouseDrop(var/atom/movable/A as mob|obj)
+	if(isliving(A) && A != usr)
+		var/mob/living/Food = A
+		if(Food.Adjacent(usr) && !stat && Food.stat != DEAD) //messy
```

Comment:
i don't see where canuseTopic would fit here. do you want me to make slime/canuseTopic() with the adjacency and stat check?